### ISSUE TITLE
isPatternAvoiding sometimes incorrect

### ISSUE THREAD

**Issue by ab2422** [Macaulay2/M2#3291]
The function `isPatternAvoiding` (and the related `avoidsAllPatterns`) from the `MatrixSchubert` package sometimes gives the incorrect output. For example:

```
i1 : needsPackage("MatrixSchubert");
i2 : isPatternAvoiding({3,1,2,4},{3,1,2})
o2 = true
i3 : isPatternAvoiding({3,1,2,4},{3,1,2,4})
o3 = true
```
Of course, sometimes it *is* also correct!
```
i4 : isPatternAvoiding({3,1,2,4},{2,1})
o4 = false
i5 : isPatternAvoiding({1,2,3,4}, {1,2,3,4})
o5 = false
```

**Comment by ab2422** [comment]
A follow-up: this may be an issue of different definitions? When I said "incorrect output" above, the definition I was using is that a permutation avoids a pattern if, in matrix notation, the permutation matrix doesn't contain the pattern's permutation matrix as a submatrix (or equivalently, that the "relative order" of the pattern never appears in the permutation).

But upon more experimenting, it seems that this code is checking for what I would think of as avoiding the inverse of the pattern? If that's intentional, then I'd like to just change my request to be a documentation clarification---the current sample pattern to avoid is it's own inverse! Could an additional sample be added where this difference in definition will be more immediately noticeable? E.g., that by this definition, 312-avoiding means that we never have i < j < k with w_k < w_i < w_j (instead of the "relative order" definition where it would mean never having w_j < w_k < w_i)

**Comment by mahrud** [comment]
Thanks Anna! Joe (@jmcdonough98) fixed this in #3353 for the next release! (It's in the development branch now)
